overlord/snapstate/snapmgr.go

```diff
@@ -215,6 +216,55 @@ func (compsu *ComponentSetup) Revision() snap.Revision {
 	return compsu.CompSideInfo.Revision
 }
 
+// ComponentSetupFromSnapSetup returns a list of ComponentSetup structs for the
+// given task. Since the task could originate from one of a few different
+// scenarios, we inspect the task for various keys to determine how to find the
+// component setups.
+//
+// The task could originate from:
+// * Installing a singular component for an already installed snap
+// * Installing/refreshing a snap with components
+// * Installing/refreshing a snap without any components
+func ComponentSetupsForTask(t *state.Task) ([]*ComponentSetup, error) {
+	switch {
+	case t.Has("component-setup") || t.Has("component-setup-task"):
+		// task comes from a component installation
+		compsup, _, err := TaskComponentSetup(t)
+		if err != nil {
+			return nil, err
+		}
+		return []*ComponentSetup{compsup}, nil
+	case t.Has("component-setups") || t.Has("component-setups-task"):
```

Comment:
I think this part is a bit premature as we do not know how we will handle the multiple components with and without snaps and the structures we will create.. Maybe better just have the TODO instead of some code.

Reply:
Sounds good, I'll add a TODO and remove them for now.